cloud: add contents about Connecting to TiDB Dedicated via Private Service Connect on Google Cloud (#14450)
pingcap/docs: tidb-cloud/set-up-private-endpoint-connections-on-google-cloud.md

---
title: Connect to a TiDB Dedicated Cluster via Private Service Connect with Google Cloud
summary: Learn how to connect to your TiDB Cloud cluster via Private Service Connect with Google Cloud.
---

# Connect to a TiDB Dedicated Cluster via Private Service Connect on Google Cloud

This document describes how to connect to your TiDB Dedicated cluster via Private Service Connect on Google Cloud.

> 

Reviewer: L2 ~ L3 用的 Private Service Connect with Google Cloud ，L6 ~ L7 用的 Private Service Connect on Google Cloud，要统一下吗

> **Tip:**
>

> > - To learn how to connect to a TiDB Dedicated cluster via private endpoint with AWS, see [Connect to TiDB Dedicated via Private Endpoint with AWS](/tidb-cloud/set-up-private-endpoint-connections.md)

Reviewer: ```suggestion
> - To learn how to connect to a TiDB Dedicated cluster via private endpoint with AWS, see [Connect to TiDB Dedicated via Private Endpoint with AWS](/tidb-cloud/set-up-private-endpoint-connections.md).
```

> - To learn how to connect to a TiDB Serverless cluster via private endpoint, see [Connect to TiDB Serverless via Private Endpoint](/tidb-cloud/set-up-private-endpoint-connections-serverless.md).

> TiDB Cloud supports highly secure and one-way access to the TiDB Cloud service hosted in a Google Cloud VPC via the [Private Service Connect](https://cloud.google.com/vpc/docs/private-service-connect). You can create an endpoint to connect to the TiDB Cloud service via the endpoint with permission.

Reviewer: 这一句看起来有点问题，通过 endpoint 创建 endpoint?
You can create an endpoint to connect to the TiDB Cloud service via the endpoint with permission.

Powered by Google Cloud Private Service Connect, the endpoint connection is secure and private, and does not expose your data to the public internet. In addition, the endpoint connection supports CIDR overlap and is easier for network management.

The architecture of the private endpoint is as follows:

![Private Service Connect architecture](/media/tidb-cloud/google-cloud-psc-endpoint-overview.png)

For more detailed definitions of the private endpoint and endpoint service, see the following Google Cloud documents:

- [Private Service Connect](https://cloud.google.com/vpc/docs/private-service-connect)
- [Access published services through endpoints](https://cloud.google.com/vpc/docs/configure-private-service-connect-services)

## Restrictions

> 

Author: ```suggestion

 - Only organization owner and the project owner can create Google Cloud Private Service Connect endpoints.
```

> - This feature is applicable to TiDB Dedicated clusters created after April 13th, 2023. For older clusters, contact support for assistance.

Author: ```suggestion
- This feature is applicable to TiDB Dedicated clusters created after April 13th, 2023. For older clusters, contact [support](/tidb-cloud/tidb-cloud-support.md) for assistance for assistance.
```

> - This feature is applicable to TiDB Dedicated clusters created after April 13th, 2023. For older clusters, contact [support](/tidb-cloud/tidb-cloud-support.md) for assistance for assistance.

Reviewer: ```suggestion
- This feature is applicable to TiDB Dedicated clusters created after April 13, 2023. For older clusters, contact [TiDB Cloud Support](/tidb-cloud/tidb-cloud-support.md) for assistance.
```

> - Only the organization owner and the project owner can create Google Cloud Private Service Connect endpoints.

Reviewer: ```suggestion
- Only the `Organization Owner` and `Project Owner` roles can create Google Cloud Private Service Connect endpoints.
```

- You can create up to 10 endpoints for each TiDB Dedicated cluster.

> - You can create up to 8 Google Cloud clusters in a project with the endpoint service configured.

Reviewer: ```suggestion
- You can create up to 8 TiDB Dedicated clusters hosted on Google Cloud in a project with the endpoint service configured.
```

- The private endpoint and the TiDB cluster must be located in the same region.
- Before you begin to create an endpoint:
    - [Enable](https://console.cloud.google.com/apis/library/compute.googleapis.com) the following APIs in your project:
        - [Compute Engine API](https://cloud.google.com/compute/docs/reference/rest/v1)
        - [Service Directory API](https://cloud.google.com/service-directory/docs/reference/rest)
        - [Cloud DNS API](https://cloud.google.com/dns/docs/reference/v1)
    - Prepare the following [IAM roles](https://cloud.google.com/iam/docs/understanding-roles) with the permissions needed to create an endpoint.

        | Task | Required IAM Roles |

>         |---|---|

Reviewer: L44 和 L55 里的 Required IAM Roles 列中的内容是相同的，那这个表格中的两行是否可以合并成一行

TiDB Cloud supports highly secure and one-way access to the TiDB Cloud service hosted in a Google Cloud VPC via the [Private Service Connect](https://cloud.google.com/vpc/docs/private-service-connect). You can create an endpoint to connect to the TiDB Cloud service via the endpoint with permission.

Powered by Google Cloud Private Service Connect, the endpoint connection is secure and private, and does not expose your data to the public internet. In addition, the endpoint connection supports CIDR overlap and is easier for network management.

The architecture of the private endpoint is as follows:

![Private Service Connect architecture](/media/tidb-cloud/google-cloud-psc-endpoint-overview.png)

For more detailed definitions of the private endpoint and endpoint service, see the following Google Cloud documents:

- [Private Service Connect](https://cloud.google.com/vpc/docs/private-service-connect)
- [Access published services through endpoints](https://cloud.google.com/vpc/docs/configure-private-service-connect-services)

## Restrictions

 - Only the organization owner and the project owner can create Google Cloud Private Service Connect endpoints.

> - You can create up to 12 endpoints for each TiDB Dedicated cluster.

Reviewer 2: ```suggestion
- You can create up to 10 endpoints for each TiDB Dedicated cluster.
```

> - The private endpoint and the TiDB cluster must be located in the same region.

Reviewer 2: - Up to 8 Google Cloud cluster can be created in a project with endpoint service configured

Author: OK，添加到这里了 https://github.com/pingcap/docs/pull/14450/files#r1299665895

- You can create up to 8 Google Cloud clusters in a project with the endpoint service configured.
- The private endpoint and the TiDB cluster must be located in the same region.

> - Before you begin to create an endpoint:

Reviewer: 这里描写的两个步骤是否可以挪到 Prerequisites 里，避免操作的时候看漏了

        - [Service Directory API](https://cloud.google.com/service-directory/docs/reference/rest)
        - [Cloud DNS API](https://cloud.google.com/dns/docs/reference/v1)
    - Prepare the following [IAM roles](https://cloud.google.com/iam/docs/understanding-roles) with the permissions needed to create an endpoint.

        | Task | Required IAM Roles |
        |---|---|
        | Create an endpoint | [Compute Network Admin](https://cloud.google.com/iam/docs/understanding-roles#compute.networkAdmin) (roles/compute.networkAdmin) and [Service Directory Editor](https://cloud.google.com/iam/docs/understanding-roles#servicedirectory.editor) (roles/servicedirectory.editor) |
        | Automatically or manually configure [DNS entries](https://cloud.google.com/vpc/docs/configure-private-service-connect-services#dns-endpoint) for an endpoint | [Compute Network Admin](https://cloud.google.com/iam/docs/understanding-roles#compute.networkAdmin) (roles/compute.networkAdmin) and [Service Directory Editor](https://cloud.google.com/iam/docs/understanding-roles#servicedirectory.editor) (roles/servicedirectory.editor) |

> - Egress firewall rules must permit traffic to the internal IP address of the endpoint. The implied allow egress firewall rule permits egress to any destination IP address.

Reviewer: ```suggestion
- Egress firewall rules must permit traffic to the internal IP address of the endpoint. The [implied allow egress firewall rule](https://cloud.google.com/firewall/docs/firewalls#default_firewall_rules) permits egress to any destination IP address.
```

> - If you have created egress deny firewall rules in your VPC network, or if you have created hierarchical firewall policies that modify the implied allowed egress behavior, access to the endpoint might be affected. You need to create a specific egress allow firewall rule or policy to permit traffic to internal IP address destination of the service endpoint.

Reviewer: ```suggestion
- If you have created egress deny firewall rules in your VPC network, or if you have created hierarchical firewall policies that modify the implied allowed egress behavior, access to the endpoint might be affected. In this case, you need to create a specific egress allow firewall rule or policy to permit traffic to the internal destination IP address of the endpoint.
```

In most scenarios, it is recommended that you use private endpoint connection over VPC peering. However, in the following scenarios, you should use VPC peering instead of private endpoint connection:

- You are using a [TiCDC](https://docs.pingcap.com/tidb/stable/ticdc-overview) cluster to replicate data from a source TiDB cluster to a target TiDB cluster across regions, to get high availability. Currently, private endpoint does not support cross-region connection.

> - You are using a TiCDC cluster to replicate data to a downstream cluster (such as Amazon Aurora, MySQL, and Kafka) but you cannot maintain the endpoint service on your own.

Reviewer: ```suggestion
- You are using a TiCDC cluster to replicate data to a downstream cluster (such as Amazon Aurora, MySQL, and Kafka) but you cannot maintain the endpoint service of the downstream on your own.
```

2. [Provide the information for creating an endpoint](#step-2-provide-the-information-for-creating-an-endpoint)
3. [Accept endpoint access](#step-3-accept-endpoint-access)
4. [Connect to your TiDB cluster](#step-4-connect-to-your-tidb-cluster)

If you have multiple clusters, you need to repeat these steps for each cluster that you want to connect to using Google Cloud Private Service Connect.

### Prerequisites

1. Log in to the [TiDB Cloud console](https://tidbcloud.com).
2. Click <MDSvgIcon name="icon-left-projects" /> in the lower-left corner, switch to the target project if you have multiple projects, and then click **Project Settings**.
3. On the **Project Settings** page of your project, click **Network Access** in the left navigation pane, and click the **Private Endpoint** tab.
4. Click **Create Private Endpoint** in the upper-right corner, and then select **Google Cloud Private Endpoint**.

### Step 1. Choose a TiDB cluster

Click the drop-down list and choose an available TiDB Dedicated cluster.

You can select a cluster with any of the following statuses:

- **Available**
- **Restoring**
- **Modifying**
- **Importing**

> - This feature is applicable to TiDB Dedicated clusters created after April 13th, 2023. For older clusters, contact support for assistance.

Author: ```suggestion

```

### Step 4: Connect to your TiDB cluster

After you have accepted the endpoint connection, take the following steps to connect to your TiDB cluster:

1. On the [**Clusters**](https://tidbcloud.com/console/clusters) page, click the name of your target cluster to go to its overview page.
2. Click **Connect** in the upper-right corner. A connection dialog is displayed.
3. Select the **Private Endpoint** tab, and then select the **Google Cloud** tab. The private endpoint you just created is displayed.
4. In the **Action** column, click **Connect**, click the tab of your preferred connection method, and then connect to your cluster with the connection string. The placeholders `<cluster_endpoint_name>:<port>` in the connection string are automatically replaced with the real values.

### Private endpoint status reference

When you use private endpoint connections, the statuses of private endpoints or private endpoint services are displayed on the [**Private Endpoint** page](#prerequisites).

The possible statuses of a private endpoint are explained as follows:

> - **Not Configured**: the endpoint service is created but the private endpoint is not created yet.

Reviewer 2: 因为这里已经改成如果没有 endpoint，那么就不会显示任何的 record，所以这里我理解可以去掉了 @Frank945946

> - **Not Configured**: the endpoint service is created but the private endpoint is not created yet.

Author: ```suggestion
```

- **Pending**: waiting for processing.
- **Active**: your private endpoint is ready to use. You cannot edit the private endpoint of this status.
- **Deleting**: the private endpoint is being deleted.
- **Failed**: the private endpoint creation fails. You can click **Edit** of that row to retry the creation.

The possible statuses of a private endpoint service are explained as follows:

- **Creating**: the endpoint service is being created, which takes 3 to 5 minutes.
- **Active**: the endpoint service is created, no matter whether the private endpoint is created or not.
- **Deleting**: the endpoint service or the cluster is being deleted, which takes 3 to 5 minutes.

## Troubleshooting

### TiDB Cloud fails to create an endpoint service, what should I do?

> Usually the endpoint service is created automatically when the cluster is created and rarely fails. If it shows as failed or remains in a **Creating** state, submit a [support ticket](/tidb-cloud/tidb-cloud-support.md) for assistance.

Reviewer 2: 目前 aws 的 endpoint service 是自动创建的，gcp 的不是，需要用户点击进入创建 gcp endpoint 的界面里，才会启动创建 endpoint service 的创建流程
![img_v2_bd819871-0d00-45d3-8c14-792976add8cg](https://github.com/pingcap/docs/assets/34148681/eed4aef0-8661-4254-ae54-f4d8f6ea2100)

> Usually the endpoint service is created automatically when the cluster is created and rarely fails. If it shows as failed or remains in a **Creating** state, submit a [support ticket](/tidb-cloud/tidb-cloud-support.md) for assistance.

Author: ```suggestion
Usually the endpoint service is created automatically when you open the **Create Google Cloud Private Endpoint** page, and the endpoint service rarely fails. If it shows as failed or remains in a **Creating** state, submit a [support ticket](/tidb-cloud/tidb-cloud-support.md) for assistance.
```

- Before you begin to create an endpoint:
    - [Enable](https://console.cloud.google.com/apis/library/compute.googleapis.com) the following APIs in your project:
        - [Compute Engine API](https://cloud.google.com/compute/docs/reference/rest/v1)
        - [Service Directory API](https://cloud.google.com/service-directory/docs/reference/rest)
        - [Cloud DNS API](https://cloud.google.com/dns/docs/reference/v1)
    - Prepare the following [IAM roles](https://cloud.google.com/iam/docs/understanding-roles) with the permissions needed to create an endpoint.

        | Task | Required IAM Roles |
        |---|---|
        | Create an endpoint | [Compute Network Admin](https://cloud.google.com/iam/docs/understanding-roles#compute.networkAdmin) (roles/compute.networkAdmin) and [Service Directory Editor](https://cloud.google.com/iam/docs/understanding-roles#servicedirectory.editor) (roles/servicedirectory.editor) |

>         | Automatically or manually configure [DNS entries](https://cloud.google.com/vpc/docs/configure-private-service-connect-services#dns-endpoint) for an endpoint | [Compute Network Admin](https://cloud.google.com/iam/docs/understanding-roles#compute.networkAdmin) (roles/compute.networkAdmin) and [Service Directory Editor](https://cloud.google.com/iam/docs/understanding-roles#servicedirectory.editor) (roles/servicedirectory.editor) |

Author: Google Cloud 也需要配置 DNS entries 吗？操作步骤中并没有相关内容。@Frank945946 PTAL

>         | Automatically or manually configure [DNS entries](https://cloud.google.com/vpc/docs/configure-private-service-connect-services#dns-endpoint) for an endpoint | [Compute Network Admin](https://cloud.google.com/iam/docs/understanding-roles#compute.networkAdmin) (roles/compute.networkAdmin) and [Service Directory Editor](https://cloud.google.com/iam/docs/understanding-roles#servicedirectory.editor) (roles/servicedirectory.editor) |

Reviewer 2: 需要的，因为你可能已经在某个 team 里了，本身就带权限了

### Private endpoint status reference

When you use private endpoint connections, the statuses of private endpoints or private endpoint services are displayed on the [**Private Endpoint** page](#prerequisites).

The possible statuses of a private endpoint are explained as follows:

- **Not Configured**: the endpoint service is created but the private endpoint is not created yet.
- **Pending**: waiting for processing.
- **Active**: your private endpoint is ready to use. You cannot edit the private endpoint of this status.
- **Deleting**: the private endpoint is being deleted.
- **Failed**: the private endpoint creation fails. You can click **Edit** of that row to retry the creation.

The possible statuses of a private endpoint service are explained as follows:

- **Creating**: the endpoint service is being created, which takes 3 to 5 minutes.
- **Active**: the endpoint service is created, no matter whether the private endpoint is created or not.

> - **Deleting**: the endpoint service or the cluster is being deleted, which takes 3 to 5 minutes.

Author: 这里没说用户可以删除 service，只是解释 service 的可能状态。如果 Cloud 在删除 service 或者相关集群，用户应该也能看到 Deleting 状态。所以这里可以保留 Deleting 状态吧？@hijoone PTAL

> - **Deleting**: the endpoint service or the cluster is being deleted, which takes 3 to 5 minutes.

Reviewer 2: > 这里没说用户可以删除 service，只是解释 service 的可能状态。如果 Cloud 在删除 service 或者相关集群，用户应该也能看到 Deleting 状态。所以这里可以保留 Deleting 状态吧？@hijoone PTAL

目前的逻辑里，客户是没办法主动删除 service 的，只有在集群被删除的情况下 service 才会被删除，而且删除集群的动作是在后台进行，console 上点一下，集群立马就不见了，所以这里我建议可以把 `Deleting` 删掉，大不了后续有变动再加

### Fail to create an endpoint in Google Cloud. What should I do?

To troubleshoot the issue, you need to review the error message returned by Google Cloud Shell after you execute the private endpoint creation command. If it is a permission-related error, you must grant the necessary permissions before retrying.

### I cancelled some actions. What should I do to handle cancellation before accepting endpoint access?

Unsaved drafts of cancelled actions will not be retained or displayed. You need to repeat each step when creating a new private endpoint in the TiDB Cloud console next time.

If you have already executed the command to create a private endpoint in Google Cloud Shell, you need to manually [delete the corresponding endpoint](https://cloud.google.com/vpc/docs/configure-private-service-connect-services#delete-endpoint) in the Google Cloud console.

### Why can't I see the endpoints generated by directly copying the service attachment in the TiDB Cloud console?

> In the TiDB Cloud console, you can only view endpoints that are created through the command generated on the **Create Google Cloud Private Endpoint** page, allowing you to see all the created endpoints within the same Google Cloud project ID.

Author: ```suggestion
In the TiDB Cloud console, you can only view endpoints that are created through the command generated on the **Create Google Cloud Private Endpoint** page, allowing you to see all the created endpoints within the Google Cloud project ID.
```

---
title: Connect to a TiDB Dedicated Cluster via Google Cloud Private Service Connect
summary: Learn how to connect to your TiDB Cloud cluster via Google Cloud Private Service Connect.
---

# Connect to a TiDB Dedicated Cluster via Google Cloud Private Service Connect

> This document describes how to connect to your TiDB Dedicated cluster via Google Cloud Private Service Connect.

Author: ```suggestion
This document describes how to connect to your TiDB Dedicated cluster via Google Cloud Private Service Connect. Google Cloud Private Service Connect is a private endpoint service provided by Google Cloud.
```

> **Tip:**
>
> - To learn how to connect to a TiDB Dedicated cluster via private endpoint with AWS, see [Connect to TiDB Dedicated via Private Endpoint with AWS](/tidb-cloud/set-up-private-endpoint-connections.md).
> - To learn how to connect to a TiDB Serverless cluster via private endpoint, see [Connect to TiDB Serverless via Private Endpoint](/tidb-cloud/set-up-private-endpoint-connections-serverless.md).

> TiDB Cloud supports highly secure and one-way access to the TiDB Cloud service hosted in a Google Cloud VPC via the [Private Service Connect](https://cloud.google.com/vpc/docs/private-service-connect). You can create an endpoint to connect to the TiDB Cloud service via the endpoint with permission.

Author: ```suggestion
TiDB Cloud supports highly secure and one-way access to the TiDB Cloud service hosted in a Google Cloud VPC via the [Private Service Connect](https://cloud.google.com/vpc/docs/private-service-connect). You can connect to the TiDB Cloud service via the endpoint with permission.
```

Powered by Google Cloud Private Service Connect, the endpoint connection is secure and private, and does not expose your data to the public internet. In addition, the endpoint connection supports CIDR overlap and is easier for network management.

The architecture of the private endpoint is as follows:

![Private Service Connect architecture](/media/tidb-cloud/google-cloud-psc-endpoint-overview.png)

For more detailed definitions of the private endpoint and endpoint service, see the following Google Cloud documents:

- [Private Service Connect](https://cloud.google.com/vpc/docs/private-service-connect)
- [Access published services through endpoints](https://cloud.google.com/vpc/docs/configure-private-service-connect-services)

## Restrictions

- This feature is applicable to TiDB Dedicated clusters created after April 13, 2023. For older clusters, contact [TiDB Cloud Support](/tidb-cloud/tidb-cloud-support.md) for assistance.
- Only the `Organization Owner` and `Project Owner` roles can create Google Cloud Private Service Connect endpoints.

> - You can create up to 10 endpoints for each TiDB Dedicated cluster.

Reviewer 2: ```suggestion
- You can create up to total 12 endpoints for each TiDB Dedicated cluster.
```

Reviewer 2: You can create up to 10 endpoints for each Google Cloud connect to a TiDB Dedicated cluster.

Reviewer 3: A TiDB cluster can handle connections from up to 12 endpoints.
Each Google Cloud project can have a maximum of 10 endpoints connecting to this TiDB Cluster.

- You can create up to 8 TiDB Dedicated clusters hosted on Google Cloud in a project with the endpoint service configured.

> - The private endpoint and the TiDB cluster must be located in the same region.

Reviewer: ```suggestion
- The private endpoint and the TiDB cluster to be connected must be located in the same region.
```

- You are using a [TiCDC](https://docs.pingcap.com/tidb/stable/ticdc-overview) cluster to replicate data from a source TiDB cluster to a target TiDB cluster across regions, to get high availability. Currently, private endpoint does not support cross-region connection.
- You are using a TiCDC cluster to replicate data to a downstream cluster (such as Amazon Aurora, MySQL, and Kafka) but you cannot maintain the endpoint service of the downstream on your own.

## Set up a private endpoint with Google Cloud Private Service Connect

To connect to your TiDB Dedicated cluster via a private endpoint, complete the [prerequisites](#prerequisites) and follow these steps:

1. [Choose a TiDB cluster](#step-1-choose-a-tidb-cluster)
2. [Provide the information for creating an endpoint](#step-2-provide-the-information-for-creating-an-endpoint)
3. [Accept endpoint access](#step-3-accept-endpoint-access)
4. [Connect to your TiDB cluster](#step-4-connect-to-your-tidb-cluster)

If you have multiple clusters, you need to repeat these steps for each cluster that you want to connect to using Google Cloud Private Service Connect.

### Prerequisites

Before you begin to create an endpoint:

> - [Enable](https://console.cloud.google.com/apis/library/compute.googleapis.com) the following APIs in your project:

Reviewer: ```suggestion
- [Enable](https://console.cloud.google.com/apis/library/compute.googleapis.com) the following APIs in your Google Cloud project:
```

- **Importing**

### Step 2. Provide the information for creating an endpoint

1. Provide the following information to generate the command for private endpoint creation:
    - **Google Cloud Project ID**: the Project ID associated with your Google Cloud account. You can find the ID on the [Google Cloud **Dashboard** page](https://console.cloud.google.com/home/dashboard).
    - **Google Cloud VPC Name**: the name of the VPC in your specified project. You can find it on the [Google Cloud **VPC networks** page](https://console.cloud.google.com/networking/networks/list).
    - **Google Cloud Subnet Name**: the name of the subnet in the specified VPC. You can find it on the **VPC network details** page.
    - **Private Service Connect Endpoint Name**: enter a unique name for the private endpoint that will be created.
2. After entering the information, click **Generate Command**.
3. Copy the command.
4. Go to [Google Cloud Shell](https://console.cloud.google.com/home/dashboard) to execute the command.

### Step 3. Accept endpoint access

After executing the command in Google Cloud Shell successfully, go back to the TiDB Cloud console and then click **Accept Endpoint Access**.

> If you see an error `No connection request received from the endpoint.`, make sure that you have copied the command correctly and successfully executed it in your Google Cloud Shell.

Reviewer 2: not received connection request from `endpoint`

> If you see an error `No connection request received from the endpoint.`, make sure that you have copied the command correctly and successfully executed it in your Google Cloud Shell.

Author: ```suggestion
If you see an error `not received connection request from endpoint`, make sure that you have copied the command correctly and successfully executed it in your Google Cloud Shell.
```

### Step 4. Connect to your TiDB cluster

After you have accepted the endpoint connection, take the following steps to connect to your TiDB cluster:

> 1. On the [**Clusters**](https://tidbcloud.com/console/clusters) page, click the name of your target cluster to go to its overview page.

Author: ```suggestion
1. On the [**Clusters**](https://tidbcloud.com/console/clusters) page, click **...** in the **Action** column.
```

> 2. Click **Connect** in the upper-right corner. A connection dialog is displayed.

Author: ```suggestion
2. Click **Connect**. A connection dialog is displayed.
```

The possible statuses of a private endpoint are explained as follows:

- **Pending**: waiting for processing.
- **Active**: your private endpoint is ready to use. You cannot edit the private endpoint of this status.
- **Deleting**: the private endpoint is being deleted.
- **Failed**: the private endpoint creation fails. You can click **Edit** of that row to retry the creation.

The possible statuses of a private endpoint service are explained as follows:

- **Creating**: the endpoint service is being created, which takes 3 to 5 minutes.
- **Active**: the endpoint service is created, no matter whether the private endpoint is created or not.

## Troubleshooting

### TiDB Cloud fails to create an endpoint service. What should I do?

The endpoint service is created automatically after you open the **Create Google Cloud Private Endpoint** page and choose the TiDB cluster. If it shows as failed or remains in the **Creating** state for a long time, submit a [support ticket](/tidb-cloud/tidb-cloud-support.md) for assistance.

> ### TiDB Cloud fails to create an endpoint. What should I do?

Reviewer 3: ```suggestion
### Fails to create an endpoint in Google Cloud. What should I do?
```

To troubleshoot the issue, you need to review the error message returned by Google Cloud Shell after you execute the private endpoint creation command. If it is a permission-related error, you must grant the necessary permissions before retrying.

### I cancelled some actions. What should I do to handle cancellation before accepting endpoint access?

Unsaved drafts of cancelled actions will not be retained or displayed. You need to repeat each step when creating a new private endpoint in the TiDB Cloud console next time.

If you have already executed the command to create a private endpoint in Google Cloud Shell, you need to manually [delete the corresponding endpoint](https://cloud.google.com/vpc/docs/configure-private-service-connect-services#delete-endpoint) in the Google Cloud console.

> ### Why cannot I see the endpoints generated by directly copying the service attachment in the TiDB Cloud console?

Author: ```suggestion
### Why can't I see the endpoints generated by directly copying the service attachment in the TiDB Cloud console?
```

Reviewer: LGTM

> In the TiDB Cloud console, you can only view endpoints that are created through the command generated on the **Create Google Cloud Private Endpoint** page, allowing you to see all the created endpoints within the same Google Cloud project ID.

Reviewer: "allowing you to see all the created endpoints within the same Google Cloud project ID." 是什么意思呢，只能看到同一个 Google Cloud project ID 的 endpoint 吗

Reviewer 2: 这里可能想表达：对于同一个  Google Cloud project 下的所有 endpoint 都可以看到

---

> title: Connect to a TiDB Dedicated Cluster via Private Service Connect with Google Cloud

Author: ```suggestion
title: Connect to a TiDB Dedicated Cluster via Google Cloud Private Service Connect
```

summary: Learn how to connect to your TiDB Cloud cluster via Private Service Connect with Google Cloud.
---

> # Connect to a TiDB Dedicated Cluster via Private Service Connect on Google Cloud

Author: ```suggestion
# Connect to a TiDB Dedicated Cluster via Private Service Connect with Google Cloud
```

> This document describes how to connect to your TiDB Dedicated cluster via Private Service Connect on Google Cloud.

Author: ```suggestion
This document describes how to connect to your TiDB Dedicated cluster via Google Cloud Private Service Connect.
```

- Egress firewall rules must permit traffic to the internal IP address of the endpoint. The implied allow egress firewall rule permits egress to any destination IP address.
- If you have created egress deny firewall rules in your VPC network, or if you have created hierarchical firewall policies that modify the implied allowed egress behavior, access to the endpoint might be affected. You need to create a specific egress allow firewall rule or policy to permit traffic to internal IP address destination of the service endpoint.

In most scenarios, it is recommended that you use private endpoint connection over VPC peering. However, in the following scenarios, you should use VPC peering instead of private endpoint connection:

- You are using a [TiCDC](https://docs.pingcap.com/tidb/stable/ticdc-overview) cluster to replicate data from a source TiDB cluster to a target TiDB cluster across regions, to get high availability. Currently, private endpoint does not support cross-region connection.
- You are using a TiCDC cluster to replicate data to a downstream cluster (such as Amazon Aurora, MySQL, and Kafka) but you cannot maintain the endpoint service on your own.

> ### Prerequisites

Reviewer: 这个  Prerequisites 要不要合并到 L72 的 Prerequisites 里

Author: 好

Before you begin to create an endpoint:

- [Enable](https://console.cloud.google.com/apis/library/compute.googleapis.com) the following APIs in your project:
    - [Compute Engine API](https://cloud.google.com/compute/docs/reference/rest/v1)
    - [Service Directory API](https://cloud.google.com/service-directory/docs/reference/rest)
    - [Cloud DNS API](https://cloud.google.com/dns/docs/reference/v1)

- Prepare the following [IAM roles](https://cloud.google.com/iam/docs/understanding-roles) with the permissions needed to create an endpoint.

    | Task | Required IAM Roles |

>     |---|---|

Reviewer: L56 和 L57 里的 Required IAM Roles 列中的内容是相同的，那这个表格中的两行是否可以合并成一行

Author: ok

4. Click **Create Private Endpoint** in the upper-right corner, and then select **Google Cloud Private Endpoint**.

### Step 1. Choose a TiDB cluster

Click the drop-down list and choose an available TiDB Dedicated cluster.

You can select a cluster with any of the following statuses:

- **Available**
- **Restoring**
- **Modifying**
- **Importing**

### Step 2. Provide the information for creating an endpoint

1. Provide the following information to generate the command to create a Google Cloud Private Service Connect endpoint:
    - **Google Cloud Project ID**: the Project ID associated with your Google Cloud account. You can find the ID on the [Google Cloud **Dashboard** page](https://console.cloud.google.com/home/dashboard).
    - **Google Cloud VPC Name**: the name of the VPC in your specified project. You can find it on the [Google Cloud **VPC networks** page](https://console.cloud.google.com/networking/networks/list).
    - **Google Cloud Subnet Name**: the name of the subnet in the specified VPC. You can find it on the **VPC network details** page.

>     - **Private Service Connect Endpoint Name**: enter a unique name for the Private Endpoint that will be created.

Reviewer: ```suggestion
    - **Private Service Connect Endpoint Name**: enter a unique name for the private endpoint that will be created.
```

2. After entering the information, click **Generate Command**.
3. Copy the command.
4. Go to [Google Cloud Shell](https://console.cloud.google.com/home/dashboard) to execute the command.

### Step 3. Accept endpoint access

> After executing the command in Google Cloud Shell, go back to the TiDB Cloud console and then click **Accept endpoint access**.

Reviewer: ```suggestion
After executing the command in Google Cloud Shell, go back to the TiDB Cloud console and then click **Accept Endpoint Access**.
```

> If you see an error "No connection request received from the endpoint.", make sure that you have copied the command correctly and successfully executed it in your Google Cloud Shell.

Reviewer: ```suggestion
If you see an error `No connection request received from the endpoint.`, make sure that you have copied the command correctly and successfully executed it in your Google Cloud Shell.
```

> ### Step 4: Connect to your TiDB cluster

Reviewer: ```suggestion
### Step 4. Connect to your TiDB cluster
```

4. In the **Action** column, click **Connect**, click the tab of your preferred connection method, and then connect to your cluster with the connection string. The placeholders `<cluster_endpoint_name>:<port>` in the connection string are automatically replaced with the real values.

### Private endpoint status reference

When you use private endpoint connections, the statuses of private endpoints or private endpoint services are displayed on the [**Private Endpoint** page](#prerequisites).

The possible statuses of a private endpoint are explained as follows:

- **Pending**: waiting for processing.
- **Active**: your private endpoint is ready to use. You cannot edit the private endpoint of this status.
- **Deleting**: the private endpoint is being deleted.
- **Failed**: the private endpoint creation fails. You can click **Edit** of that row to retry the creation.

The possible statuses of a private endpoint service are explained as follows:

- **Creating**: the endpoint service is being created, which takes 3 to 5 minutes.
- **Active**: the endpoint service is created, no matter whether the private endpoint is created or not.

## Troubleshooting

> ### TiDB Cloud fails to create an endpoint service, what should I do?

Reviewer: ```suggestion
### TiDB Cloud fails to create an endpoint service. What should I do?
```

> The endpoint service is created automatically when you open the **Create Google Cloud Private Endpoint** page. If it shows as failed or remains in the **Creating** state for a long time, submit a [support ticket](/tidb-cloud/tidb-cloud-support.md) for assistance.

Reviewer: 确认下，是打开  **Create Google Cloud Private Endpoint** page 页面时 endpoint service 就会自动创建吗，还是选择集群后呢？

Reviewer 2: 选择集群后，endpoint service 是集群粒度的，一个集群一个

> ### TiDB Cloud fails to create an endpoint, what should I do?

Reviewer: ```suggestion
### TiDB Cloud fails to create an endpoint. What should I do?
```

> You need to review the error message generated during execution in Google Cloud Shell to troubleshoot the issue. If it is a permission-related error, you must grant the necessary permissions before retry.

Reviewer: ```suggestion
To troubleshoot the issue, you need to review the error message returned by Google Cloud Shell after you execute the private endpoint creation command. If it is a permission-related error, you must grant the necessary permissions before retrying.
```

### I cancelled some actions. What should I do to handle cancellation before accepting endpoint access?

> Unsaved drafts of cancelled actions will not be retained or displayed. You need to reconfigure each step when creating a new private endpoint in the TiDB Cloud console next time.

Reviewer: ```suggestion
Unsaved drafts of cancelled actions will not be retained or displayed. You need to repeat each step when creating a new private endpoint in the TiDB Cloud console next time.
```

If you have already executed the command to create a private endpoint in Google Cloud Shell, you need to manually [delete the corresponding endpoint](https://cloud.google.com/vpc/docs/configure-private-service-connect-services#delete-endpoint) in the Google Cloud console.

> ### Why can't I see the endpoints generated by directly copying the service attachment in the TiDB Cloud console?

Reviewer: ```suggestion
### Why cannot I see the endpoints generated by directly copying the service attachment in the TiDB Cloud console?
```

> You can view endpoints created through the command generated in the TiDB Cloud console, allowing you to see all the created endpoints within the same Google Cloud project ID.

Reviewer: ```suggestion
In the TiDB Cloud console, you can only view endpoints that are created through the command generated on the **Create Google Cloud Private Endpoint** page, allowing you to see all the created endpoints within the same Google Cloud project ID.
```
